packages/dev/s2-docs/src/getPages.ts

```diff
@@ -0,0 +1,88 @@
+import {cache} from 'react';
+import {getBaseUrl} from './pageUtils';
+import {glob, readFile} from 'fs/promises';
+import type {Page} from '@parcel/rsc';
+// eslint-disable-next-line
+import {transformAsync} from '@parcel/rust/lib/index.js';
+
+export const getPages = cache(async () => {
```

Comment:
This parses all MDX files so we have a complete list of pages, even when the Parcel build doesn't include all of them (building only S2 or RAC). Using an internal Parcel utility here for performance (the pages get parsed in parallel in Rust). Will have to keep this up to date with Parcel upgrades. Ideally there'd be a public utility for this.